### DISCUSSION TITLE
Importing Modules

### DISCUSSION THREAD

**Discussion by troyam** [ruby2js/ruby2js#52] [Q&A]
Sorry, I'm learning Ruby and I'm trying to use your module to convert to js.
The conversion works fine with filter --strict.

First case: 

I'm trying to convert some files and I'm getting the code looks:

`require(File.expand_path("../boot", __FILE__))`;

File now is a function and I'm getting `"File is not defined"`;

My question is how to load submodules( in ruby ) of these .rb codes...

Can you explain to me please? Thanks!

**Comment by rubys** [comment]
The answer would depend on your use case.

1.  Are you trying to get an existing code base to run pretty much as is in the context of a browser or some JavaScript environment?  If so, you might want to look into [opal](https://opalrb.com/) instead.
2. Are you trying to migrate your code from Ruby to JavaScript?  If so, be aware that JavaScript doesn't have exact equivalents to everything that Ruby has, so at best the conversion will be 80 to 95% successful.  The statement above likely would map to an `import` statement, or perhaps a CJS require, but will require some manual editing to complete.  Additionally, JavaScript has no native equivalent to `File.expand_path`, the closest it has is `path.resolve` which is roughly equivalent to Ruby's `File.absolute_path`.
3. Are you trying to make use of an platform API that typically is only made available in JavaScript but you want to leverage your existing skills in Ruby?  This use case differs significantly from the previous two in that you generally aren't starting with an existing code base.
